Tutorials/EB/GeometryGeneration/main.cpp

```diff
@@ -8,59 +8,111 @@ using namespace amrex;
 
 int main (int argc, char* argv[])
 {
-    amrex::Initialize(argc, argv);
+  amrex::Initialize(argc, argv);
 
+  {
+    int n_cell = 128;
+    int max_grid_size = 32;
+
+    // read parameters
     {
-        int n_cell = 128;
-        int max_grid_size = 32;
+      ParmParse pp;
+      pp.query("n_cell", n_cell);
+      pp.query("max_grid_size", max_grid_size);
+    }
 
-        // read parameters
-        {
-            ParmParse pp;
-            pp.query("n_cell", n_cell);
-            pp.query("max_grid_size", max_grid_size);
-        }
+    Geometry geom;
+    {
+      RealBox rb({-5.2,-5.2,-1.0}, {5.2,5.2,2.0}); // physical domain
+      Array<int,AMREX_SPACEDIM> is_periodic{false, false, false};
+      Geometry::Setup(&rb, 0, is_periodic.data());
+      Box domain(IntVect(0), IntVect(n_cell-1));
+      geom.define(domain);            
+    }
 
-        Geometry geom;
-        {
-            RealBox rb({-0.5,-0.5,-0.5}, {0.5,0.5,0.5}); // physical domain
-            Array<int,AMREX_SPACEDIM> is_periodic{false, false, false};
-            Geometry::Setup(&rb, 0, is_periodic.data());
-            Box domain(IntVect(0), IntVect(n_cell-1));
-            geom.define(domain);            
-        }
+    {
```

Comment:
Could you keep the old setup and add the new one as an additional example?  Maybe you can add an inputs ParmParse parameter for the geometry type.